azkaban-web-server/src/main/java/azkaban/scheduler/QuartzScheduler.java

```diff
@@ -117,18 +121,20 @@ public void unregisterJob(final String groupName) throws SchedulerException {
   }
 
   /**
-   * Only cron schedule register is supported.
+   * Only cron schedule register is supported. Since register might be called when
+   * concurrently uploading projects, so synchronized is added to ensure thread safety.
    *
    * @param cronExpression the cron schedule for this job
    * @param jobDescription Regarding QuartzJobDescription#groupName, in order to guarantee no
    * duplicate quartz schedules, we design the naming convention depending on use cases: <ul>
-   * <li>User flow schedule: we use {@link org.quartz.JobKey#JobKey} to represent the identity of a
+   * <li>User flow schedule: we use {@link JobKey#JobKey} to represent the identity of a
    * flow's schedule. The format follows "$projectID_$flowName" to guarantee no duplicates.
-   * <li>Quartz schedule for AZ internal use: the groupName should start with letters,
-   * rather than
+   * <li>Quartz schedule for AZ internal use: the groupName should start with letters, rather
```

Comment:
Update the comments for groupName definition which should be "projectID.flowName" and start with number?

Reply:
will do thanks.